### PR TITLE
GG-33016 [IGNITE-14402] Java thin client: Continuous queries support

### PR DESCRIPTION
**Pull request by ptupitsyn**
Co-authored-by: Aleksey Plekhanov <plehanov.alex@gmail.com>

### CODE REVIEW

**Review by isapego — CHANGES_REQUESTED** [review]
See my comments inline

File: modules/core/src/main/java/org/apache/ignite/client/ClientDisconnectListener.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/client/ClientDisconnectListener.java]: Fixed

File: modules/core/src/main/java/org/apache/ignite/client/ClientDisconnectListener.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+/**
+ * Client disconnected event listener. Such listeners can be used in {@link ClientCache#query(ContinuousQuery,
+ * ClientDisconnectListener)} or {@link ClientCache#registerCacheEntryListener(CacheEntryListenerConfiguration,
+ * ClientDisconnectListener)} methods to handle client channel failure.
+ */
+public interface ClientDisconnectListener {
+    /**
+     * Client disconnected callback.
+     *
+     * @param reason Exception that caused the disconnect, can be {@code null}.
```
[the remark this answers is not in the document]
Comment (ptupitsyn) [reply modules/core/src/main/java/org/apache/ignite/client/ClientDisconnectListener.java:31]: Done

**Review by isapego — APPROVED** [review]
Looks good to me

**Pull request merged**